client/src/routes/home/expense/AddExpense.tsx

```diff
@@ -1,5 +1,145 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import Expense from "../../../types/Expense";
+import { FormEvent, useEffect, useState } from "react";
+import { Link, redirect, useNavigate, useParams } from "react-router-dom";
+import { toast } from "@/hooks/use-toast";
+import { X } from "lucide-react";
+
 const AddExpense = () => {
-  return <div>AddExpense</div>;
+  const queryClient = useQueryClient();
+  const { category } = useParams();
+  const nav = useNavigate();
+
+  const [name, setName] = useState("");
+  const [price, setPrice] = useState(1.0);
+  const [quantity, setQuantity] = useState(1);
+  const [total, setTotal] = useState(0);
+  
+
+  useEffect(() => {
+    setTotal(price * quantity);
+  }, [price, quantity]);
+
+  const { mutate, isPending } = useMutation({
+    mutationFn:async(e: FormEvent) => {
+      e.preventDefault()
+
+      if (!name.trim()) {
+        throw Error("Name is required.")  
+      }
+
+      if (price <= 0) {
+        throw Error("Price must be greater than zero.") 
+      }
+
+      if (quantity <= 0) {
+        throw Error("Quantity must be greater than zero.")
+      }
+      
+      const expense: Expense = {
+        expense_name: name,
+        price: price,
+        quantity: quantity,
+        total: total,
+        category_id: Number(category)
+      }
+      await fetch('http://localhost:3000/expense', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(expense)
+      })
+    },
+    onSuccess: () => {
+      toast({
+        description: "Expense Added!"
+      })
```

Comment:
@cassandraysabel Hi pa add:
```ts
queryClient.invalidateQueries({queryKey: ["category", category]})
```

its used to refetch the expenses of that category

Reply:
ayaw mag push ako nalang ma push